pkg/packagekit/autoupdates.jsx

```diff
@@ -213,14 +213,125 @@ class DnfImpl extends ImplBase {
+                        this.enabled = (output.indexOf("enabled\n") >= 0);
+                        this.type = (output.indexOf("security\n") >= 0) ? "security" : "all";
+
+                        // if we have OnCalendar=, use that (we disable OnUnitInactiveSec= in our drop-in)
+                        const calIdx = output.indexOf("OnCalendar=");
+                        if (calIdx >= 0) {
+                            this.parseCalendar(output.substr(calIdx).split('\n')[0].split("=")[1]);
+                        } else {
+                            if (output.indexOf("InactiveSec=1d\n") >= 0)
+                                this.day = this.time = "";
+                            else if (this.installed)
+                                this.supported = false;
+                        }
+
+                        debug(`apt getConfig: supported ${this.supported}, enabled ${this.enabled}, type ${this.type}, day ${this.day}, time ${this.time}, installed ${this.installed}; raw response '${output}'`);
+                        resolve();
+                    })
+                    .catch(error => {
+                        console.error("apt getConfig failed:", error);
+                        this.supported = false;
+                        resolve();
+                    });
+        });
+    }
+
+    setConfig(enabled, type, day, time) {
+        const timerConf = "/etc/systemd/system/apt-daily-upgrade.timer.d/time.conf";
+        let script = "set -e; ";
+        const cockpitConf = "/etc/apt/apt.conf.d/60cockpit";
+
+        // if we enable through Cockpit, make sure that starting the timer doesn't start the .service right away,
+        // due to the packaged default OnBootSec=1h; just set a reasonable initial time which will trigger the code below
+        if (enabled && !this.enabled && !this.time && !this.day)
+            time = "6:00";
+
+        if (time !== null || day !== null) {
+            if (day === "" && time === "") {
+                // restore defaults
+                script += "rm -f " + timerConf + "; ";
+            } else {
+                if (day == null)
+                    day = this.day;
+                if (time == null)
+                    time = this.time;
+                script += "mkdir -p /etc/systemd/system/apt-daily-upgrade.timer.d; ";
+                script += "printf '[Timer]\\nOnBootSec=\\nOnCalendar=" + day + " " + time + "\\n' > " + timerConf + "; ";
+                script += "systemctl daemon-reload; ";
+            }
+        }
+
+        if (enabled !== null) {
+            script += "systemctl " + (enabled ? "enable" : "disable") + " --now apt-daily-upgrade.timer; ";
+
+            if (enabled) {
+                if (type !== null) {
+                    script += "echo 'Unattended-Upgrade::Automatic-Reboot \"true\";\n' > " + cockpitConf + ";";
+
+                    // Only security updates is the default on Debian/Ubuntu
```

Comment:
I don't think so. In debian-stable, it seems to do everything:
```
        "origin=Debian,codename=${distro_codename},label=Debian";
        "origin=Debian,codename=${distro_codename},label=Debian-Security";
        "origin=Debian,codename=${distro_codename}-security,label=Debian-Security";
```

Same for Ubuntu:
```
        "${distro_id}:${distro_codename}";
        "${distro_id}:${distro_codename}-security";
```

My Debian server runs unattended-upgrades, and it installs everything. So the code below needs adjusting, and also handle Debian and Ubuntu separately.